src/Components/firebase-access.jsx

```diff
@@ -3,465 +3,386 @@ import "firebase/auth";
+    let IPOyearMax = 0;
+    let today = new Date();
+    let year = today.getFullYear();
+    Stocks.forEach(function(Stock){
+        if (IPOyearMax < Stock.data().IPOyear){
+            IPOyearMax = Stock.data().IPOyear;
+        }
+    });
 
-// requests prices and technical indicator images to be written to the database
-export const initializeQuiz = async (symbols, roomId, periodLen, endDates) => {
-  var formData = new FormData();
-  formData.append('symbol',JSON.stringify(symbols));
-  formData.append('RoomId',roomId);
-  formData.append('end-date',JSON.stringify(endDates));
-  try{
-      await fetch('http://localhost:8080/get-prices', {
-          method: 'POST',
-          mode: 'cors',
-          body: formData
-      })
-  }
-  catch(err) {
-      console.log("Error is " +  err)
-  }
-  formData.append('periodLen',periodLen)
-  try{
-      await fetch('http://localhost:8080/get-stock-image', {
-          method: 'POST',
-          mode: 'cors',
-          body: formData
-      })
-  }
-  catch(error){
-      console.log("Error is " +  error)
-  }
+    // No more than 7 rounds(Periods are measured in months)
+    let min_window_size = 3;
+    let yearDiff = year - (IPOyearMax+1);
+    let maximum_period = Math.floor(((yearDiff * 12)  - min_window_size) / Rounds);
+    let random_period =  Math.floor((Math.random()  * (maximum_period - min_window_size))+min_window_size);
+    let startDate = new Date(IPOyearMax+1,1,1);
+    let endDate =  new Date(IPOyearMax+1,1+random_period,1);
+    let rand_startDate =  randomDate(startDate,endDate);
+    let dates = [];
+    let curr_date = rand_startDate;
+    for(var i = 0; i < Rounds; i++) {
+        dates.push(curr_date.toISOString().substring(0, 10));
+        curr_date = new Date(curr_date.setMonth(curr_date.getMonth()+random_period));
+    }
+    const datesD ={
+        "dates" : dates,
+        "period": random_period
+    }
+    return datesD
 }
 
-// gets a list of all symbols being tracked
-export const getSymbols = async (db, roomID) => {
-  const symbol = await db.collection('Rooms').doc(roomID).get();
-  const symbolData = await symbol.data();
-  return symbolData.symbols;
+export const initializeQuiz = async (symbols, roomId, periodLen, endDates) => {
+    var formData = new FormData();
+    formData.append('symbol',JSON.stringify(symbols));
+    formData.append('RoomId',roomId);
+    formData.append('end-date',JSON.stringify(endDates));
+    try{
+        await fetch('http://localhost:8080/get-prices', {
+            method: 'POST',
+            mode: 'cors',
+            body: formData
+        })
+    }
+    catch(err) {
+        console.log("Error is " +  err)
+    }
+    formData.append('periodLen',periodLen)
+    try{
+        await fetch('http://localhost:8080/get-stock-image', {
+            method: 'POST',
+            mode: 'cors',
+            body: formData
+        })
+    }
+    catch(error){
+        console.log("Error is " +  error)
+    }
```

Comment:
use proper try catch style
```suggestion
    try {
        await fetch('http://localhost:8080/get-prices', {
            method: 'POST',
            mode: 'cors',
            body: formData
        })
    } catch(err) {
        console.log("Error is " +  err)
    }
    formData.append('periodLen',periodLen)
    try {
        await fetch('http://localhost:8080/get-stock-image', {
            method: 'POST',
            mode: 'cors',
            body: formData
        })
    } catch(error){
        console.log("Error is " +  error)
    }
```